### ISSUE TITLE
ApiException in mpclarkson/freshdesk-php-sdk/src/Exceptions/ApiException.php:56

### ISSUE THREAD

**Issue by snipe** [mpclarkson/freshdesk-laravel#2]
Hi there - is this package still maintained? (I only ask since it's been a while since there have been recent commits, and there aren't many issues.)

I'm getting an error with a basic test:

```
 try {
            $me = Freshdesk::agents()->current();
            print_r($me);

        } catch (\Exception $e) {
            echo 'ERROR: '.$e;
        }
```

The error I get is below:

```
ERROR:Freshdesk\Exceptions\ApiException in /Users/agianotto/Sites/snipe-it/grokability-site/vendor/mpclarkson/freshdesk-php-sdk/src/Exceptions/ApiException.php:56
Stack trace:
#0 /Users/agianotto/Sites/snipe-it/grokability-site/vendor/mpclarkson/freshdesk-php-sdk/src/Api.php(260): Freshdesk\Exceptions\ApiException::create(Object(GuzzleHttp\Exception\RequestException))
#1 /Users/agianotto/Sites/snipe-it/grokability-site/vendor/mpclarkson/freshdesk-php-sdk/src/Api.php(227): Freshdesk\Api->performRequest('GET', 'https://support...', Array)
#2 /Users/agianotto/Sites/snipe-it/grokability-site/vendor/mpclarkson/freshdesk-php-sdk/src/Resources/Agent.php(54): Freshdesk\Api->request('GET', '/agents/me', NULL, NULL)
#3 /Users/agianotto/Sites/snipe-it/grokability-site/app/Console/Commands/SyncFreshdesk.php(94): Freshdesk\Resources\Agent->current()
#4 [internal function]: App\Console\Commands\SyncFreshdesk->handle()
#5 /Users/agianotto/Sites/snipe-it/grokability-site/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(29): call_user_func_array(Array, Array)
#6 /Users/agianotto/Sites/snipe-it/grokability-site/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(87): Illuminate\Container\BoundMethod::Illuminate\Container\{closure}()
#7 /Users/agianotto/Sites/snipe-it/grokability-site/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(31): Illuminate\Container\BoundMethod::callBoundMethod(Object(Illuminate\Foundation\Application), Array, Object(Closure))
#8 /Users/agianotto/Sites/snipe-it/grokability-site/vendor/laravel/framework/src/Illuminate/Container/Container.php(539): Illuminate\Container\BoundMethod::call(Object(Illuminate\Foundation\Application), Array, Array, NULL)
#9 /Users/agianotto/Sites/snipe-it/grokability-site/vendor/laravel/framework/src/Illuminate/Console/Command.php(182): Illuminate\Container\Container->call(Array)
#10 /Users/agianotto/Sites/snipe-it/grokability-site/vendor/symfony/console/Command/Command.php(264): Illuminate\Console\Command->execute(Object(Symfony\Component\Console\Input\ArgvInput), Object(Illuminate\Console\OutputStyle))
#11 /Users/agianotto/Sites/snipe-it/grokability-site/vendor/laravel/framework/src/Illuminate/Console/Command.php(167): Symfony\Component\Console\Command\Command->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Illuminate\Console\OutputStyle))
#12 /Users/agianotto/Sites/snipe-it/grokability-site/vendor/symfony/console/Application.php(869): Illuminate\Console\Command->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#13 /Users/agianotto/Sites/snipe-it/grokability-site/vendor/symfony/console/Application.php(223): Symfony\Component\Console\Application->doRunCommand(Object(App\Console\Commands\SyncFreshdesk), Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#14 /Users/agianotto/Sites/snipe-it/grokability-site/vendor/symfony/console/Application.php(130): Symfony\Component\Console\Application->doRun(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#15 /Users/agianotto/Sites/snipe-it/grokability-site/vendor/laravel/framework/src/Illuminate/Foundation/Console/Kernel.php(122): Symfony\Component\Console\Application->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#16 /Users/agianotto/Sites/snipe-it/grokability-site/artisan(35): Illuminate\Foundation\Console\Kernel->handle(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
```

I have filled out the freshdesk.php config file, and can confirm that laravel is able to read the confog correctly. This is on Laravel 5.4. Any ideas? Thanks so much!

**Comment by mpclarkson** [comment]
Hi there, yes it is maintained theoretically but I'm not doing much PHP at the moment... Happy of you do fork and submit a PR, otherwise I'll try and have a look on the weekend.

**Comment by snipe** [comment]
Aha - I think I figured it out. We use a custom subdomain, and it doesn't seem to like that. When I use our old `blah.freshdesk.com` domain, it works, when I use `support.blah.com`, it bonks hard. When I dump the response, I noticed this:

 `[host:GuzzleHttp\Psr7\Uri:private] => support.blah.com.freshdesk.com`

Which I think is the issue.

This gets a little tricky only because once you change your subdomain in Freshdesk, I don't *think* they provide a place for you to see what the original `*.freshdesk.com` URL was when you signed up.

At any rate, through trial and error I figured it out, and it looks to be working using the original `blah.freshdesk.com` domain. 

I'll see if I have time this weekend to patch the SDK library to handle that use case, or perhaps at least update the documentation to make it clear to use the `*.freshdesk.com` domain.

Are you possibly interested in some documentation PRs to flesh out the verbs a little more? 

Thanks for the package! Guzzle makes REST easy, but this wrapper is even nicer :)
